src/utils/db/car_stream.rs

```diff
@@ -44,6 +44,7 @@ impl CarBlock {
     pub fn write(&self, mut writer: &mut impl io::Write) -> io::Result<()> {
         let frame_length = self.cid.encoded_len() + self.data.len();
         writer.write_all(&frame_length.encode_var_vec())?;
+        #[allow(clippy::needless_borrows_for_generic_args)]
```

Comment:
is clippy wrong?

Reply:
I'm inclined it's the API design to blame. `self.cid.write_bytes(&mut writer)` unneccesarily takes the ownership here